pkg/loadbalancer/maps/pressure_metrics.go

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package maps
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/job"
+	"github.com/cilium/statedb"
+
+	"github.com/cilium/cilium/pkg/loadbalancer"
+	"github.com/cilium/cilium/pkg/metrics"
+)
+
+type pressureMetricsParams struct {
+	cell.In
+
+	JobGroup        job.Group
+	MetricsRegistry *metrics.Registry
+	DB              *statedb.DB
+	Frontends       statedb.Table[*loadbalancer.Frontend]
+	Backends        statedb.Table[*loadbalancer.Backend]
+	Config          loadbalancer.Config
+	LBMaps          LBMaps
+}
+
+type pressureMetrics struct {
+	pressureMetricsParams
+
+	bpfMaps *BPFLBMaps
+	gauges  map[string]*metrics.GaugeWithThreshold
+}
+
+func registerPressureMetricsReporter(p pressureMetricsParams) {
+	if !p.Config.EnableExperimentalLB || p.Config.LBPressureMetricsInterval == 0 {
+		return
+	}
+
+	bpfMaps, ok := p.LBMaps.(*BPFLBMaps)
+	if !ok {
+		return
+	}
+
+	pm := &pressureMetrics{
+		pressureMetricsParams: p,
+		gauges:                map[string]*metrics.GaugeWithThreshold{},
+		bpfMaps:               bpfMaps,
+	}
+
+	// Update the metrics only once every 5 minutes as we need to iterate over all
+	// keys in each map to extract the count. If we're testing update the metrics
+	// frequently.
+	updateInterval := p.Config.LBPressureMetricsInterval
+	p.JobGroup.Add(job.Timer(
+		"pressure-metrics-reporter",
+		pm.report,
+		updateInterval,
+	))
+}
+
+func (pm *pressureMetrics) getGauge(mapName string) *metrics.GaugeWithThreshold {
+	mapName = strings.TrimPrefix(mapName, metrics.Namespace+"_")
+	if g, found := pm.gauges[mapName]; found {
+		return g
+	}
+	// Create a new metric. This will be registered/unregistered on-demand based on
+	// the threshold. E.g. if map pressure is 0.0 it won't appear in metrics.
+	g := pm.MetricsRegistry.NewBPFMapPressureGauge(mapName, 0.0)
+	pm.gauges[mapName] = g
+	return g
+}
+
+func (pm *pressureMetrics) report(ctx context.Context) error {
+	var errs error
+
+	// openMaps is immutable after Start(), so safe to access here.
+	for name, m := range pm.bpfMaps.openMaps {
+		if ctx.Err() != nil {
+			break
+		}
+		if count, err := m.Count(); err != nil {
```

Comment:
The CT map BPF pressure metric uses batch ops to count the elements in the map. Curious why the same approach is not taken here

Reply:
Ah wasn't aware of this! I'm not using `pkg/bpf` so no access to `NewBatchIterator` or `CountAll` though. Could look at whether we can make those available when using plain `ebpf.Map`. Or I could also consider switching `pkg/loadbalancer/maps` over to `pkg/bpf`.

Reply:
I'm going to switch to `pkg/bpf` as `pkg/ebpf` is being removed and will consider using the batch ops for this.

Reply:
https://github.com/cilium/cilium/pull/39210

Will wiat for that to merge and will then update this PR.

Reply:
Hmm, BatchIterator requires concrete types but I don't have them in `pressure_metrics.go` so I can't instantiate it. I'll look at maybe doing a `(*bpf.).Count` method that uses batch lookups underneath instead.